### DISCUSSION TITLE
Can we add multiple content sources per PnP - Search Results?

### DISCUSSION THREAD

**Discussion by gretchunkim** [microsoft-search/pnp-modern-search#2868] [Q&A] [answered]
Is it possible to add multiple MS Search (or SharePoint) data sources as long as they are the same data source type (all Microsoft Search or all SharePoint Search)? I wasn't sure if it's by design (plural "Content sources").  

Receiving API call error message, but not sure that's because I'm being greedy. 

Objective: Trying to display  segments of data in multiple tables (non-linear).  For pure space saving purpose. 
![image](https://user-images.githubusercontent.com/8564035/226803762-48ea6f73-7a86-4b6f-b565-42159cbb8fbc.png)

![image](https://user-images.githubusercontent.com/8564035/226803748-f9fd459a-df10-4985-8265-b661eae9ff8c.png)

![image](https://user-images.githubusercontent.com/8564035/226803937-ae6c8847-6284-468a-be70-e706a253214f.png)

**Comment by wobba** [comment, marked as answer]
You can add an `*` there to pick all connections you have. I'll tag this as an improvement to support a comma separated list of multiple connections.

**Reply by gretchunkim** [reply]
Thank you @wobba . Looking up data binding method for multi data source.

**Comment by renewtx** [comment]
Hello,

I am trying to figure out  how/where you actually specify the content sources?  Per the above screenshot snippets there appears to be a Content Sources input field, but I don't see that  ... I am using v4.8 of the PnP Search.  Is this a NEW feature not publicly available yet?  Here is what I see ...

![image](https://github.com/microsoft-search/pnp-modern-search/assets/58831734/14bf3d09-453c-4264-81ef-5a18e4114876)

Thanks

**Reply by kasperbolarsen** [reply]
Content sources will only show up if you have selected ExternalItem in the Entity Types field

![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/417ad449-f775-4cab-91a3-3087782181a1)

**Reply by renewtx** [reply]
Perfect!  Thanks, that did it.  

Follow up -- is it possible to have a single Search Result web part show results from multiple content sources?  For example, if I want to show results from both a file share and SharePoint?  It sounds like based on the above if I enter a '*' in Content Sources, it should use results from all, but does that include a mix if sources such as file shares and SharePoint?  Or do the Content Sources all have to be of the same type?  If possible, are you aware of any documentation on how to properly configure this type of setup?

**Reply by gretchunkim** [reply]
@renewtx  it's one single source type and not mixing types. You could use MS Graph API for looking up SP contents though. Documentation will be mixture of MS Graph documentation and PnP query templates section will be helpful to you.
